src/ray/core_worker/task_event_buffer.cc

```diff
@@ -89,16 +89,29 @@ void TaskStatusEvent::ToRpcTaskEvents(rpc::TaskEvents *rpc_task_events) {
   if (state_update_->error_info_.has_value()) {
     dst_state_update->set_error_type(state_update_->error_info_->error_type());
   }
+  return false;
 }
 
-void TaskProfileEvent::ToRpcTaskEvents(rpc::TaskEvents *rpc_task_events) {
+bool TaskProfileEvent::ToRpcTaskEventsOrDrop(rpc::TaskEvents *rpc_task_events) {
+  // Rate limit on the number of profiling events from the task. This is especially the
+  // case if a driver has many profiling events when submitting tasks
+  auto profile_events = rpc_task_events->mutable_profile_events();
+  auto profile_event_max_num =
+      RayConfig::instance().task_events_max_num_profile_events_for_task();
+  if (profile_event_max_num >= 0 &&
```

Comment:
hmm why don't we only drop the extra profile events, but we still report the task event?

Reply:
This is only dropping the profile events. A single `TaskEvent` captures a single profile event, where the `rpc_task_events` is a collection of status events + profile events to be sent.

Reply:
ah I see. Can you comment that from the code?